### PR TITLE
Adding yaml file for SCIM2 APIs 

### PR DESCRIPTION
**Pull request by sarubi**
Adding .yaml file for the current SCIM2 APIs.

### CODE REVIEW

File: scim2.yaml [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+  description: SCIM 2.0 endpoints
+  version: '1.0'
+  title: SCIM 2.0 Endpoint Swagger Definition
+  contact:
+    name: WSO2 Identity Server Team
+    url: 'http://wso2.com'
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0'
+basePath: /scim2/
```
Comment (omindu) [inline scim2.yaml]: /t/{tenant-domain}/scim2

Comment (sarubi) [reply scim2.yaml]: fixed

File: scim2.yaml [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+        - Me Endpoint
+      summary: Delete the authenticated user.
+      description: Returns HTTP 204 if the user is successfully deleted.
+      operationId: deleteUserMe
+      x-wso2-request: |
+         curl -v -k --user  [username]:[password] -X DELETE https://localhost:9443/scim2/Me
+      x-wso2-curl: |
+         curl -v -k --user kim:kimwso2 -X DELETE https://localhost:9443/scim2/Me
+      x-wso2-response: |
+        HTTP/1.1 501 Not Implemented
```
Comment (omindu) [inline scim2.yaml]: Is this correct ?

Comment (sarubi) [reply scim2.yaml]: Yes, we didn't implement 'deleteMe' function yet, but the endpoint is there.
https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/a4883981e661ee6122ee50405aef350f1eaeae0c/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java#L598

https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/a4883981e661ee6122ee50405aef350f1eaeae0c/components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/MeResource.java#L131

**Pull request merged**